### ISSUE TITLE
Increase slug size limit beyond 500MB

### ISSUE THREAD

**Issue by friism** [heroku/roadmap#196]
### Required Terms

- [X] I agree to follow this project's [Code of Conduct](https://github.com/heroku/roadmap/blob/main/CODE_OF_CONDUCT.md)
- [X] I have read and accept the [Salesforce Program Agreement](https://www.salesforce.com/company/program-agreement/)

### What service(s) is this request for?

Runtime, Builds

### Tell us about what you're trying to solve. What challenges are you facing?

Currently Heroku limits slug size to 500MB (although it can be increased on a per-app basis to 650MB by contacting support). This is limiting for customers that for one reason or another need to pack more stuff into their slugs. A similar limit is _not_ in place for container images (an alternative on Heroku to slugs).

The limit was put in place with good intentions, chiefly to ensure that dynos can boot quickly so apps can scale fast and so that commands like `heroku run` feel instantaneous (fetching slugs from storage is a non-trivial part of dyno boot time).

Computer performance (including for Heroku dynos) has increased a lot since the limit was last increased to 500MB in 2016 so we could probably safely increase the limit pending internal validation and testing.

For Heroku customers reading this, please comment with details on how the current limit affects you. And in particular we're curious to learn what's driving slug-size up for you. Is it assets (images), lots of dependencies or something else? We want to help you fit whatever you need into your slugs, but we're also curious to learn what you want to put in there so we can help in the best way.

**Comment by Neel49** [comment]
It really sucks. Trying to put object detection models in

**Comment by PolpoDevs** [comment]
> For Heroku customers reading this, please comment with details on how the current limit affects you. And in particular we're curious to learn what's driving slug-size up for you. 

One concrete example: we use some parts of LibreOffice, and the barebones install of this is roughly 300MB, the soft limit for the slug size. Add to that the regular dependencies we already have to run the code that interacts with it, and we exceed it easily.

While we are still below the 500MB limit, we are working on features that could theoretically be run perfectly fine by the same dynos. However, due to the slug size, we will need to spread these across multiple projects, or even use a completely different service altogether to not run into the slug size limit.

The biggest question I've heard amongst the team is why the slug size does not scale along with the type of dyno, or perhaps even an optional flag (perhaps even at some additional cost). In fact, I came across this roadmap / issue by accident while looking for information on this as I wanted to make sure it is indeed the case. From all I could find, it seems regular 1X dynos are limited in slug size in the exact same way the highest level performance dynos are. Is there any explanation for this?

**Comment by ryan-sherpa** [comment]
Same issue here. We've had to roll separate applications to reduce the slug size, including specific buildpacks in each. 

We really run into this issue with our use of Review Apps, where we can't easily deploy multiple applications and thus need all buildpacks in a single app. We simply cannot run our entire app in these Review App environments. 

Sounds like containerization will be the way to go, but wish there was a simpler fix in the short-term.

**Comment by shiladityab24** [comment]
Yes slug size needs to increased more than 500MB. Faced the same issue for the spring boot apps in java while deploying the jar file , whose size exceeded more than 500MB. It would be very helpful if the limit of the slug size is increased.Need to perform Repackaging the jar files where lot of libraries were removed. Especially for the libraries which are related to the jasper report.

**Comment by capeterson** [comment]
We're at a bit of a fork in the road in terms of decisions around this. On one hand, we have plans for making it possible to mount S3 on heroku dynos as a filesystem (with the normal caveats that the S3 APIs aren't especially performant at random writes in the middle of a file etc), which seems like it would be a better option for models and static assets and plays into some of our plans to make static web apps easier, as well as making it easier to work with persisting files in the context of the ephemeral filesystems dynos have. 

The other option is to of course just increase slug sizes directly, but that means a redeployment for new static assets etc that's otherwise not really justified and doesn't help with uploading or content that needs to change more often than app deployments. 

For the sake of validating some of our thinking I'd love some feedback on how much the s3 mounting solution helps, or doesn't, for some of your use cases. If it doesn't help then knowing more about what kind of sizes you need to target and for what use case would be very handy.

**Comment by tmm1** [comment]
In our case with a big rails app, just the combined rubygems and node_modules + compiled assets for a complex frontend js app gets us very close to 500MB. As useful as an S3 mount would be, I think the current limit of 500MB is too low and a source of friction.

**Comment by friism** [comment]
Hey, the default slug size is now 1GB. Let us know if this helps: https://devcenter.heroku.com/changelog-items/3612
